openshift/openshift

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+
+if [ "x$RADARGUN_HOME" = "x" ]; then
+  DIRNAME=`dirname $0`
+  RADARGUN_HOME=`cd $DIRNAME/..; pwd`
+fi;
+export RADARGUN_HOME
+
+OSE_MAIN_VERSION=v3.7.1
+OSE_SHA1_VERSION=ab0f056
+TEST_PROJECT=myproject
+LOGIN_COMMAND=""
+RADARGUN_IMAGE=radargun
+FULL_IMAGE=${DOCKER_REGISTRY}/${TEST_PROJECT}/${RADARGUN_IMAGE}
+RADARGUN_MASTER="radargun-master.${TEST_PROJECT}.svc"
+CONFIG_DIR=configs/
+RADARGUN_CONFIG="library-dist-reads.xml"
+DEFAULT_JAVA_OPTS="-XX:+UseG1GC -verbose:gc -XX:MaxGCPauseMillis=300 -XX:InitiatingHeapOccupancyPercent=70 -XX:+PrintGCDateStamps -XX:+PrintGCDetails"
+CUSTOM_JAVA_OPTS=""
+TOTAL_CONTAINER_MEM_SLAVE=1024
+NUMBER_OF_SLAVES=2
+
+JVM_NATIVE_MB=80
+
+help_and_exit() {
+  echo "Usage: "
+  echo '  $ openshift [OPTIONS] COMMAND'
+  echo ""
+  echo "COMMAND:"
+  wrappedecho "  login" "Login to OpenShift."
+  wrappedecho '  newproject' 'Create new project within OpenShift. This is be default named "myproject".'
+  wrappedecho "  build" "Build and push Docker image to OpenShift Docker registry."
+  wrappedecho "  deploy" "Deploy RadarGun to OpenShift."
+  wrappedecho "  results" "Download RadarGun logs and benchmark results from OpenShift to localhost."
+  wrappedecho "  purge" "Purge OpenShift project. All RadarGun related objects will be removed from the OpenShift project."
+  wrappedecho ""
+  echo "OPTIONS:"
+  wrappedecho '  -L LOGIN_COMMAND' 'Login command specific to your OpenShift provider. Defaults to empty string. Must start with oc. Applicable to login command.'
+  wrappedecho "  -cd CONFIG_DIR" "Directory with confiration files for RadarGun and its plugins. Defaults to configs/. Applicable to deploy command."
+  wrappedecho "  -cf CONFIG_FILE" "RadarGun benchmark file. Must be placed in CONFIG_DIR. Defaults to library-dist-reads.xml. Applicable to deploy command."
+  wrappedecho "  -s NUMBER_OF_SLAVES" "Number of RadarGun slaves. Defaults to 2. Applicable to deploy command."
+  wrappedecho "  -j CUSTOM_JAVA_OPTS" "Custom JAVA_OPTS that will be passed to the slave and master process JVM. Applicable to deploy command."
+  wrappedecho "  -m TOTAL_CONTAINER_MEM_SLAVE" "Total container memory for Slave containers. Defaults to 1024MiB. Applicable to deploy command."
+  exit 0
+}
+
+function wrappedecho {
+  printf '%-25s' "${1}"; echo "${2}"
+}
+
+function download_oc_client {
+  if [ -f ./oc ]; then
+    echo "OC client installed"
+  else
+    echo "Installing OC Client ..."
```

Comment:
This is assuming that the platform is Linux. It may be good to put a check in to make sure. You don't have to support MacOS!

Reply:
Thanks for the tip and the review in general! I've pushed one more commit with the assertion.